prefect_aws/ecs.py

```diff
@@ -0,0 +1,1019 @@
+
+
+def get_container(containers: List[dict], name: str) -> Optional[dict]:
+    """
+    Extract the a container from a list of containers or container definitions
+    If not found, `None` is returned.
+    """
+    for container in containers:
+        if container.get("name") == name:
+            return container
+    return None
+
+
+class ECSTask(Infrastructure):
+    """
+    <span class="badge-api experimental"/>
+
+    Run a command as an ECS task.
+
+    Note this block is experimental. The interface may change without notice.
+    """
+
+    _block_type_slug = "ecs-task"
+    _block_type_name = "ECS Task"
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/1jbV4lceHOjGgunX15lUwT/db88e184d727f721575aeb054a37e277/aws.png?h=250"  # noqa
+
+    type: Literal["ecs-task"] = Field(
+        "ecs-task", description="The slug for this task type."
+    )
+
+    aws_credentials: AwsCredentials = Field(
+        title="AWS Credentials",
+        default_factory=AwsCredentials,
+        description="The AWS credentials to use to connect to ECS.",
+    )
+
+    # Task definition settings
+    task_definition_arn: Optional[str] = Field(
+        default=None,
+        description=(
+            "An identifier for an existing task definition to use. If fields are set "
+            "on the `ECSTask` that conflict with the task definition, a new copy "
+            "will be registered with the required values. "
+            "Cannot be used with `task_definition`. If not provided, Prefect will "
+            "generate and register a minimal task definition."
+        ),
+    )
+    task_definition: Optional[dict] = Field(
+        default=None,
+        description=(
+            "An ECS task definition to use. Prefect may set defaults or override "
+            "fields on this task definition to match other `ECSTask` fields. "
+            "Cannot be used with `task_definition_arn`. If not provided, Prefect will "
+            "generate and register a minimal task definition."
+        ),
+    )
+    image: Optional[str] = Field(
+        default_factory=get_prefect_image_name,
+        description=(
+            "The image to use for the Prefect container in the task. If this value is "
+            "not null, it will override the value in the task definition. This value "
+            "defaults to a Prefect base image matching your local versions."
+        ),
+    )
+    auto_deregister_task_definition: bool = Field(
+        default=True,
+        description=(
+            "If set, any task definitions that are created by this block will be "
+            "deregistered. Existing task definitions linked by ARN will never be "
+            "deregistered. Deregistering a task definition does not remove it from "
+            "your AWS account, instead it will be marked as INACTIVE."
+        ),
+    )
+
+    # Mixed task definition / run settings
+    cpu: int = Field(
+        title="CPU",
+        default=None,
```

Comment:
Should this be:
```suggestion
        default=ECS_DEFAULT_CPU,
```

Reply:
No. We may find a value in the existing task definition and need to know if it should be overridden or not.